pkg/prometheus/promcfg.go

```diff
@@ -2223,3 +2235,92 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	}
 	return cfg
 }
+
+func (cg *ConfigGenerator) generateTracingConfig(tracingConfig *v1.PrometheusTracingConfig) (yaml.MapItem, error) {
+	cfg := yaml.MapSlice{}
+	objMeta := cg.prom.GetObjectMeta()
+
+	if tracingConfig.ClientType != "" {
```

Comment:
Since you've added a validation with enums in Kubebuilder, there is no need to check clientType here. Kubernetes API will refuse Prometheus resources if clientType is not set to grpc or http

Reply:
Changed 😄